pkg/resources/management.cattle.io/v3/project/validator.go

```diff
@@ -94,20 +100,33 @@ func (a *admitter) admitDelete(project *v3.Project) (*admissionv1.AdmissionRespo
 	return admission.ResponseAllowed(), nil
 }
 
-func (a *admitter) admitCreateOrUpdate(oldProject, newProject *v3.Project) (*admissionv1.AdmissionResponse, error) {
-	fieldErr, err := a.checkClusterName(newProject)
+func (a *admitter) admitCreate(project *v3.Project) (*admissionv1.AdmissionResponse, error) {
+	fieldErr, err := a.checkClusterExists(project)
 	if err != nil {
 		return nil, fmt.Errorf("error checking cluster name: %w", err)
 	}
 	if fieldErr != nil {
 		return admission.ResponseBadRequest(fieldErr.Error()), nil
 	}
+	return a.admitCommonCreateUpdate(nil, project)
+}
+
+func (a *admitter) admitUpdate(oldProject, newProject *v3.Project) (*admissionv1.AdmissionResponse, error) {
+	if oldProject.Spec.ClusterName != newProject.Spec.ClusterName {
+		fieldErr := field.Invalid(projectSpecFieldPath.Child("clusterNameField"), newProject.Spec.ClusterName, "field is immutable")
```

Comment:
Should this be the following?
```suggestion
		fieldErr := field.Invalid(projectSpecFieldPath.Child("clusterName"), newProject.Spec.ClusterName, "field is immutable")
```

Reply:
Or, rather, the constant `clusterNameField` as elsewhere?

Reply:
My bad, good catch.

Reply:
Fixed.